src/acceptor.rs

```diff
@@ -149,7 +148,15 @@ impl rustls_acceptor {
     /// out_accepted: An output parameter. The pointed-to pointer will be set
     ///   to a new rustls_accepted only when the function returns
     ///   RUSTLS_RESULT_OK. The memory is owned by the caller and must eventually
-    ///   be freed.
+    ///   be freed
+    /// out_alert: An output parameter. The pointed-to pointer will be set
+    ///   to a new rustls_accepted_alert only when the function returns
+    ///   a non-OK result. The memory is owned by the caller and must eventually
```

Comment:
```suggestion
    /// out_alert: An output parameter. The pointed-to pointer will be set
    ///   to a new rustls_accepted_alert when, and only when, the function returns
    ///   a non-OK result. The memory is owned by the caller and must eventually
```

Note: this guarantee that it's always set is complicated by the fact that AcceptedAlert::new can error. It looks like that error is necessitated by the Write interface, but can't actually happen since we're writing to a Vec.

Normally I'd say "assert that there's no error writing to the Vec and panic if that's not true," but panicking isn't an option for us in FFI land. So I'd handle the (can't happen) error case by making an empty AceptedAlert.

One other possibility: the upstream `AcceptedAlert` doesn't actually expose a bytes accessor, it exposes a [`write`](https://docs.rs/rustls/latest/rustls/server/struct.AcceptedAlert.html#method.write) method that expects a `&mut dyn Write`. In rustls-ffi our equivalent of that is a [`rustls_write_callback`](https://docs.rs/rustls-ffi/latest/rustls_ffi/io/type.rustls_write_callback.html) plus a `void *userdata`. And it's likely the user already has that callback handy to write to whatever their underlying I/O is. What do you think of exposing the `write` method on AcceptedAlert instead of the byte accessor?

Reply:
> What do you think of exposing the write method on AcceptedAlert instead of the byte accessor?

Yeah! That actually works out nicely and lets us drop the awkward `AcceptedAlertBuffer` type I had to invent before.

I've updated the branch, the docs, and the PR description to be arranged around using a `rustls_write_callback` for this job too. Thanks for the suggestion :+1: